### PR TITLE
[FLINK-15290][hive] Need a way to turn off vectorized orc reader for SQL CLI

### PR DESCRIPTION
**Pull request by lirui-apache**
## What is the purpose of the change

Load configurations in `HiveTableInputFormat` to decide whether vector orc reader should be used

## Brief change log

  - Call `GlobalConfiguration.loadConfiguration` when creating `HiveTableInputFormat`
  - Add test for the vector reader config

## Verifying this change

New test case

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

NA

### CODE REVIEW

**Comment by JingsongLi** [comment]
Thanks @lirui-apache , LGTM.

**Comment by JingsongLi** [comment]
CC: @KurtYoung

**Comment by JingsongLi** [comment]
Can you change title?

**Comment by JingsongLi** [comment]
I think we should not `GlobalConfiguration.loadConfiguration` in `RichInputFormat.configure`.
We should get this `TABLE_EXEC_HIVE_FALLBACK_MAPRED_READER` option in client side, must have `FLINK_CONF_DIR` in client side.
In `HiveTableSource.getDataStream`, we can get the option and pass it to `HiveTableInputFormat`.

**Comment by lirui-apache** [comment]
@JingsongLi Updated to read the configurations on client side.

File: flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/connectors/hive/read/HiveTableInputFormat.java [hunk 1]
```diff
@@ -103,17 +105,16 @@ public HiveTableInputFormat(
 		this.jobConf = new JobConf(jobConf);
 		int rowArity = catalogTable.getSchema().getFieldCount();
 		selectedFields = projectedFields != null ? projectedFields : IntStream.range(0, rowArity).toArray();
+		useMapRedReader = GlobalConfiguration.loadConfiguration().getBoolean(HiveOptions.TABLE_EXEC_HIVE_FALLBACK_MAPRED_READER);
```
Comment (JingsongLi) [inline flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/connectors/hive/read/HiveTableInputFormat.java]: You can only pass `useMapRedReader` here.
We already have `GlobalConfiguration.loadConfiguration()` in `tableSource.getDataStream`. We can just get `useMapRedReader` and pass it to input format. It is good to reduce this unhappy codes.

Comment (lirui-apache) [reply flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/connectors/hive/read/HiveTableInputFormat.java]: Done

**Pull request closed without merging**